gradle-plugins/biz.aQute.bnd.gradle/src/main/java/aQute/bnd/gradle/BndPlugin.java

```diff
@@ -252,11 +251,7 @@ public void execute(Task tt) {
 				sourceSet.getOutput()
 					.setResourcesDir(destinationDir);
 				TaskProvider<AbstractCompile> compileTask = tasks.named(sourceSet.getCompileJavaTaskName(),
-					AbstractCompile.class, t -> {
-						t.getDestinationDirectory()
-							.fileValue(destinationDir);
-						jarLibraryElements(t, sourceSet.getCompileClasspathConfigurationName());
```

Comment:
Removing this is wrong. It causes Bnd workspace builds to fail to see the jars on the classpath when building at is only sees the output folders. Bnd workspace builds (i.e. BndPlugin) need to be able to configure the compile classpath configuration to use jars and not output folders.  This is true for both compilation as well as bundle assembly. In Bnd, a project can build a jar which contains classes which are not built by the project's compilation step (using `-conditionalpackage`, `Export-Package` for example) and so when a bundle depends upon another project, the compile path must utilize the jar built by the other project. Otherwise compilation errors can occur due to types which are not visible in the project's output directory. 

So any solution also needs to address the actual compile classpath used during compilation and not just during jar assembly.

In non-Bnd workspace builds, (using BundleTaskExtension), Bnd is not responsible for managing the compile classpath. BundleTaskExtension only needs a classpath for jar assembly which is defaulted to the main compile classpath.

Reply:
I've updated this PR with a possible fix for this which doesn't modify the existing Gradle configurations, but ensures the jar are on the compilation classpaths used by the tasks.

Do you have a test case where requesting only classes would break and jars are required to pass that I could test this against?  This passes `:gradle-plugins:build`.

Also, the instructions in the README suggest running `:build` and `:publish`, but these tasks don't exist in the root project.  This looks like an oversight.

Reply:
IIUC, just the class files are no longer appropriate.  In that case, we should just move `jarLibraryElements` outside the block that configures the task so that it's always done regardless if the compile task is configured or not.  That would mean you can't use the task in `jarLibraryElements`. It looks like it's just for logging, so I think that's OK.